Nama Lengkap : Nazarul Bagus Riyadi

NIM : 240401010229

Kelas : IF404

---

# Praktikum Pertemuan 2: Struktur Data Python, NumPy & Pandas


In [1]:
import pandas as pd
import numpy as np

url = 'https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv'
df = pd.read_csv(url)
df.head()


   PassengerId  Survived  Pclass                                                 Name     Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked
0            1         0       3                              Braund, Mr. Owen Harris    male  22.0      1      0         A/5 21171   7.2500   NaN        S
1            2         1       1  Cumings, Mrs. John Bradley (Florence Briggs Thayer)  female  38.0      1      0          PC 17599  71.2833   C85        C
2            3         1       3                               Heikkinen, Miss. Laina  female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S
3            4         1       1         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1      0            113803  53.1000  C123        S
4            5         0       3                             Allen, Mr. William Henry    male  35.0      0      0            373450   8.0500   NaN        S

In [2]:
# Q1 - jumlah baris, kolom, dan nama kolom
print(f"Jumlah baris: {df.shape[0]}")
print(f"Jumlah kolom: {df.shape[1]}")
print("Nama kolom:")
for col in df.columns:
    print(f"- {col}")


Jumlah baris: 891
Jumlah kolom: 12
Nama kolom:
- PassengerId
- Survived
- Pclass
- Name
- Sex
- Age
- SibSp
- Parch
- Ticket
- Fare
- Cabin
- Embarked


In [3]:
# Q2 - missing value dan proporsi
missing_values = df.isnull().sum()
missing_percentage = (df.isnull().mean()) * 100
missing_data = pd.DataFrame({
    'Jumlah Missing Values': missing_values,
    'Proporsi Missing Values (%)': missing_percentage.round(2)
})
missing_data = missing_data[missing_data['Jumlah Missing Values'] > 0]
print(missing_data)


          Jumlah Missing Values  Proporsi Missing Values (%)
Age                         177                        19.87
Cabin                       687                        77.10
Embarked                      2                         0.22


In [4]:
# Q3 - persen penumpang yang selamat
survived_count = df['Survived'].sum()
total_count = df.shape[0]
survived_percentage = (survived_count / total_count) * 100
print(f"Persentase penumpang yang selamat: {survived_percentage:.2f}%")


Persentase penumpang yang selamat: 38.38%


In [5]:
# Q4 - penumpang wanita kelas 1
female_class_1 = df[(df['Pclass'] == 1) & (df['Sex'] == 'female')]
print(f"Jumlah: {len(female_class_1)}, Rata-rata usia: {female_class_1['Age'].mean():.2f}")


Jumlah: 94, Rata-rata usia: 34.61


In [6]:
# Q5 - tingkat keselamatan per kelas
survival_rate = df.groupby('Pclass')['Survived'].mean()
print(f"Tingkat keselamatan per kelas:\n{survival_rate}")
print(f"Kelas dengan tingkat keselamatan tertinggi: Kelas {survival_rate.idxmax()}")


Tingkat keselamatan per kelas:
Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64
Kelas dengan tingkat keselamatan tertinggi: Kelas 1


In [7]:
# Q6 - analisis tambahan jenis kelamin
survival_by_gender = df.groupby('Sex')['Survived'].mean()
print(survival_by_gender)


Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64


## Kesimpulan

### Apa yang Dipelajari
- Membaca dataset CSV dengan **Pandas** (`read_csv`, `head`, `info`, `describe`)
- Seleksi dan filter data dengan boolean indexing
- Grouping dan agregasi dengan `groupby()`
- Operasi dasar **NumPy** pada array

### Temuan Utama
- Dataset Titanic: 891 penumpang, 12 kolom — banyak missing di `Age` dan `Cabin`
- Penumpang kelas 1 lebih tinggi survival rate-nya dibanding kelas 3
- Perempuan memiliki tingkat kelangsungan hidup jauh lebih tinggi (~74% vs ~19%)

### Keterbatasan
- Analisis masih eksploratif, belum ada visualisasi atau statistik inferensial
- Missing values belum ditangani — baru diidentifikasi
